EXT_BeamPhysics.md

```diff
@@ -0,0 +1,269 @@
+
+- `macroCharge/`
+    - Type: Optional *(float)*
+    - Description: "Macro"-particles are simulation particles that represent multiple real particles.
+    The `macroCharge` is the the total charge of the collection of particles that a macro-particle
+    represents.
+
+- `momentum/`
+    - Type: Optional 2 or 3-vector *(float)*
+    - Description: The total momentum of the particles relative to the `momentumOrigin` attribute.
+    - Components: (`px`, `py`) or (`px`, `py`, `pz`).
+    - Note: A program using phase space coordinates to describe particle positions will typically use the transverse momentum
+    `px` and `py` along with the `totalMomentum` or `energy` records.
+
+- `pathLength/`
+    - Type: Optional *(float)*
+    - Description: Length that a particle has traveled.
+
+- `position`
+    - Type: Required 3-vector *(float)*
+    - Description: Position relative to the coordinate origin.
+    - Components: (`x`, `y`, `<z-coord>`) where `<z-coord>` is one of:
+        - `z`: True longitudinal coordinate.
+        - `-beta.c.dt`: Phase space coordinate conjugate to a momentum based "pz".
+        - `-beta.c.t`:Phase space coordinate conjugate to a momentum based "pz".
+        - `-c.dt`: Phase space coordinate conjugate to an energy based "pz".
+        - `-c.t`: Phase space coordinate conjugate to an energy based "pz".
+        - Where: **beta** = particle speed, **c** = speed of light, **t** = time, and **dt** = time relative to the reference time.
+
+- `refTime/`
+    - Type: Optional *(float)*
+    - Description: The reference particle time.
+    Note that the reference time is a function of **s** and therefore can be different for different particles.
+
+- `s-Position`
+    - Type: Optional *(float)*
+    - Description: Absolute longitudinal distance in **lattice** coordinates.  
+
+- `speed/`
+    - Type: Optional *(float)*
+    - Description: The speed of the particles.
+
+- `spin/`
+    - Type: Optional 3-vector *(float)*
+    - Description: Particle spin.
+    - Components: (`x`, `y`, `s`) or (`r`, `theta`, `phi`).
+
+- `time/`
+    - Type: Optional *(float)*
+    - Description: Absolute particle time. Note: Particles may have different times if the snapshot
+    is, for example, taken at constant **s**.
+
+- `totalMomentum/`
+    - Type: Optional *(float)*
+    - Description: The total momentum of the particles relative to the `referenceTotalMomentum` attribute.
+
+Non Per-Particle Records of the `Particle Root Group`
+---------------------
+
+The following possible records of the `Particle Root Group` are for specifying properties of the entire group of particles.
+
+- `phaseSpaceFirstOrderMoment/`
+  - Type: Optional 6-vector *(float)*
+  - Description: First order beam moments.
+
+- `phaseSpaceSecondOrderMoment/`
+  - Type: Optional 6x6-matrix *(float)*
+  - Description: Second order beam moments.
+
+- `phaseSpaceThirdOrderMoment/`
+  - Type: Optional 6x6x6-tensor *(float)*
+  - Description: Third order beam moments.
```

Comment:
Same questions on the order of the 6 components.

Reply:
Yes it should be mentioned that (x, px, y, py, z, pz) should be used.

Reply:
In the upcoming revision this information has been put in the **Definitions** section